src/main/java/edu/umich/ctools/analytics/AnalyticsServlet.java

```diff
@@ -0,0 +1,329 @@
+	
+	public void init() {
+		M_log.debug("init(): called");
+		getExternalAppProperties();
+	}
+	
+	private void getExternalAppProperties() {
+		M_log.debug("getExternalAppProperties(): called");
+		String propertiesFilePath = System.getProperty(SYSTEM_PROPERTY_FILE_PATH);
+		if (!ReportUtilities.isEmpty(propertiesFilePath)) {
+			appExtPropertiesFile=ReportUtilities.getPropertiesFromURL(propertiesFilePath);
+			if(appExtPropertiesFile!=null) {
+				canvasToken = appExtPropertiesFile.getProperty(PROPERTY_CANVAS_ADMIN);
+				canvasURL = appExtPropertiesFile.getProperty(PROPERTY_CANVAS_URL);
+			}else {
+				M_log.error("Failed to load application properties from canvasReport.properties");
+			}
+			
+		}else {
+			M_log.error("File path for (canvasReport.properties) is not provided");
+		}
+		
+	}
+	
+
+	protected void doGet(HttpServletRequest request,HttpServletResponse response){
+		M_log.debug("doGet: Called");
+		if(request.getParameter(ACTION).equals(ACTION_GET_COURSES)) {
+			getPublishedCourses(request,response);
+		}else if(request.getParameter(ACTION).equals(ACTION_GET_TERMS)) {
+			enrollmentTermsLogic(response);
+		}
+		
+	}
+	private void getPublishedCourses(HttpServletRequest request,HttpServletResponse response) {
+		M_log.debug("getPublishedCourses: Called");
+		String term = request.getParameter(TERM);
+		String subAccountUrl = canvasURL+"/api/v1/accounts/1/sub_accounts?recursive=true&per_page=100";
+		String url = canvasURL+"/api/v1/accounts/1/courses?enrollment_term_id="+term+"&published=true&per_page=100";
+		CoursesForSubaccounts cfs =new CoursesForSubaccounts();
+		getSubAccountInfo(subAccountUrl,cfs);
+		getThePublishedCourseList(url,cfs);
+		response.setContentType("text/csv");
+		response.setHeader("Content-Disposition","attachment;filename=publishedCourseByTerm.csv");
+		try {
+			OutputStream outputStream = response.getOutputStream();
+			String outputResult = generateCSVFile(cfs).toString();
+			outputStream.write(outputResult.getBytes());
+			outputStream.flush();
+			outputStream.close();
+		} catch (IOException e) {
+			M_log.error("Writing the csv file to the has some errror: ",e);
+		}
+	}
+	protected void doPost(HttpServletRequest request,HttpServletResponse response) {
+		M_log.debug("doGet: Called");
+	}
+
+	private StringBuilder generateCSVFile(CoursesForSubaccounts cfs) {
+		M_log.debug("generateCSVFile: Called");
+		StringBuilder writer = new StringBuilder();
+		ArrayList<Course> courses = cfs.getCourse();
+		writer.append("COURSE_NAME");writer.append(',');
+		writer.append("COURSE_ID");writer.append(',');
+		writer.append("COURSE_URL");writer.append(',');
+		writer.append("INSTRUCTOR_NAME");writer.append(',');
+		writer.append("INSTRUCTOR_EMAIL");writer.append(',');
+		writer.append("ACCOUNT_NAME");writer.append(',');
+		writer.append("PARENT_ACCOUNT_NAME");
+		writer.append('\n');
+		int count=0;
+		for (Course course : courses) {
+			writer.append("\"" +course.getCourseName()+ "\"");writer.append(',');
```

Comment:
Course should have an instance method to generate this as a string.  This method should call that method for every course and deliver the string to the writer.